app/src/main/java/to/dev/dev_android/util/network/NetworkUtils.kt

```diff
@@ -0,0 +1,29 @@
+package to.dev.dev_android.util.network
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.net.InetSocketAddress
+import java.net.Socket
+
+typealias NetworkStatusCallback = (isOnline: Boolean) -> Unit
+
+object NetworkUtils {
+    suspend fun isOnline(): Boolean = withContext(Dispatchers.IO) {
+        try {
+            val timeout = 1500
+            val address = InetSocketAddress("8.8.8.8", 80)
```

Comment:
Is it possible to check for `BuildConfig.baseUrl` instead? I know this is a lower level Socket call but if the website is down and a reconnect is possible at some point once it's available again we would be killing two birds with one stone.

Reply:
I like the way you think ( : Yes, that'll be definitely possible. I'll fix that.

Reply:
Apparently, `BuildConfig.baseUrl` doesn't work. It shouldn't be a url, but a hostname. Therefore, instead of programmatically adding `removePrefix` and `removeSuffix`, I added a `baseHostname` to the gradle properties, and then added it to the build config generator.